BoardPanel.java

```diff
@@ -0,0 +1,66 @@
+import java.awt.*;
+import java.awt.event.*;
+import java.util.ArrayList;
+import javax.swing.*;
+
+public class BoardPanel extends JPanel {
+    private MineSweeper mineSweeper;
+    private int numRows;
+    private int numCols;
+    private MineTile[][] board;
+    private ArrayList<MineTile> mineList;
+
+    public BoardPanel(MineSweeper mineSweeper, int numRows, int numCols, MineTile[][] board, ArrayList<MineTile> mineList) {
```

Comment:
Make a constructor that take all the variables inside it to initialize the Board Panel